src/bootstrap/lib.rs

```diff
@@ -253,6 +253,10 @@ pub struct Build {
     ci_env: CiEnv,
     delayed_failures: RefCell<Vec<String>>,
     prerelease_version: Cell<Option<u32>>,
+    tool_artifacts: RefCell<HashMap<
+        Interned<String>,
+        HashMap<String, (&'static str, PathBuf, Vec<String>)>
```

Comment:
Please either document these fields here or make this a struct instead of a tuple.